### DISCUSSION TITLE
Audio Segmentation Before The Inference

### DISCUSSION THREAD

**Discussion by mertbkts** [flashlight/flashlight#576] [Q&A]
Hello. I'm trying to use ASR from flashlight and i probably will work on long audio files, like 3 to 5 minutes. And i want to segment them before the inference because of less processing time. I read your "MLS: A Large-Scale Multilingual Dataset for Speech Research" paper and it was saying that you segmented the audio files into 10-20 segments when there was no speech. And you pointed that you used wav2letter@anywhere framework for it. I tried interactive streaming from it but it was crashing after a while. Then i found out wav2letter moved into flashlight. But i couldn't find exactly how i can segment my audio files. I tried the VAD tool from flashlight but it was taking much longer time than the actual inference. I thought maybe i can segment them evenly to 10-20 second parts, but i don't want to take the risk that cutting a word in half in the speech. So do you have any suggestion of tools or a library that i can use to segment my long audio files to cut them into 10-20 second segments, or what did you use to segment your files? I will be very grateful for any help.

The paper that i talked about: https://arxiv.org/abs/2012.03411

**Comment by tlikhomanenko** [comment]
cc @vineelpratap

**Reply by sciai-ai** [reply]
Any updates on this? In an facing the same issue
